### PR TITLE
Remove-exceptions: make lambda types explicit

### PR DESCRIPTION
**Pull request by smowton**
Small cleanup brought in from comments on an out-of-tree PR.

### CODE REVIEW

File: src/goto-programs/remove_exceptions.cpp [hunk 1]
```diff
@@ -570,10 +571,10 @@ void remove_exceptions(
   const namespacet ns(symbol_table);
   std::map<irep_idt, std::set<irep_idt>> exceptions_map;
   uncaught_exceptions(goto_functions, ns, exceptions_map);
-  // NOLINTNEXTLINE
-  auto function_may_throw = [&exceptions_map](const irep_idt &id) {
-    return !exceptions_map[id].empty();
-  };
+  remove_exceptionst::function_may_throwt function_may_throw =
+    [&exceptions_map](const irep_idt &id) { // NOLINT(*)
```
Comment (tautschnig) [inline src/goto-programs/remove_exceptions.cpp]: I'd suggest to be explicit and use `NOLINT(whitespace/braces)`

**Review by tautschnig — COMMENTED** [review]
I think that's a good idea, but I'm still curious what your reason for this patch was?

**Comment by smowton** [comment]
Reviewer on the other PR noted these were using `auto` when the type wasn't otherwise stated, and better compiler complaints would result if I was explicit.

**Comment by tautschnig** [comment]
Maybe you can add as body to the commit message:
"These were using auto when the type wasn't otherwise stated, and better compiler complaints would result if the type is made explicit."

**Review by tautschnig — APPROVED** [review]
It would be nice to clarify the "why" in the commit message as discussed, but otherwise that's good of course.

**Comment by smowton** [comment]
done

**Pull request merged**